packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -218,20 +218,20 @@ export const useStoreSwitcher = (props = {}) => {
                             `/${storeCode}`
                         )}${params}`;
 
-                        window.location.assign(newPath);
+                        globalThis.location.assign(newPath);
```

Comment:
I'd be nervous about taking on this scope, but should react-router handle this? I'm positive I missed some justification in the demo why this was needed, but just wanted to check. Some added benefits would also being able to derive path with `useLocation` and not have to craft our own paths.

```js
const history = useHistory()
history.replace(newPath)
```

Reply:
In general, you're correct: where possible, we should use React Router's `history` and `location` APIs to read from and write to the URL. The problem with store switching is that we designate a leading store code (e.g., `fr`) as a `basename`, and those APIs (e.g., `history.push()`) presume we aren't changing the `basename`. Since this code's purpose is to change the `basename`, it needs to use the browser's `location.assign()` instead.

That said, I ensured that code in this file only accesses `window.location` from within a user-interaction callback, `handleSwitchStore`. The server won't run this code, I believe.